storage_scrubber/src/checks.rs

```diff
@@ -47,6 +48,39 @@ impl TimelineAnalysis {
     }
 }
 
+fn check_valid_layermap(metadata: &HashMap<LayerName, LayerFileMetadata>) -> Option<String> {
```

Comment:
Maybe incorporate the PR description of a valid layer map into a doc comment for the function? This definition of layer validity could change later.

Reply:
Also wondering if you could point me to the relevant compaction PR that guarantees this layer map structure. Thx!

Reply:
For the record, the validation code in the compaction.rs is this one: https://github.com/neondatabase/neon/blob/0f6568426342f80c6faba81af9827cc4ad9fa2d6/pageserver/src/tenant/timeline/compaction.rs#L1932-L1962

It's not trivial to share the code in this case, but do we have any ideas how we'll ensure this code will be in sync in the future? Maybe extract compaction.rs part into a separate function, then link to that function from here and vice versa?

Reply:
The piece of code is implemented in 3 places now: scrubber, compaction, and unit test force layer creation. I plan to unify them in a follow up PR. Basically all we need is the layer name to determine the validity, and all these 3 places can collect the layer name structs.